### PR TITLE
Implementation of a new cgmes boundary server

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-stream-binder-rabbit</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-cassandra</artifactId>
+        </dependency>
+        <dependency>
```
Comment (geofjamg) [inline pom.xml:148]: Duplicated dependency

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-swagger-ui</artifactId>
+            <version>${springfox.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-swagger2</artifactId>
+            <version>${springfox.version}</version>
+        </dependency>
+        <dependency>
```
Comment (geofjamg) [inline pom.xml:163]: Do you really need this one?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-triple-store-impl-rdf4j</artifactId>
+            <version>${powsybl-core.version}</version>
+            <scope>runtime</scope>
+        </dependency>
+
+        <dependency>
```
Comment (geofjamg) [inline pom.xml:193]: The 2 following dependency are not runtime and should be in the previous section

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-stream</artifactId>
+        </dependency>
+        <dependency>
```
Comment (geofjamg) [inline pom.xml:140]: Do you need this one?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+        </dependency>
+
+        <!-- Test dependencies -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${junit.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (geofjamg) [inline pom.xml:211]: Do you need this one?

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+            <groupId>com.github.nosan</groupId>
+            <artifactId>embedded-cassandra-spring-test</artifactId>
+            <version>${embedded-cassandra.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
```
Comment (geofjamg) [inline pom.xml:231]: Do you need this one?

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    private BoundaryRepository boundaryRepository;
+
+    public CgmesBoundaryService(BoundaryRepository boundaryRepository) {
+        this.boundaryRepository = boundaryRepository;
+    }
+
+    Optional<BoundaryInfo> getBoundary(String boundaryId) {
+        Optional<BoundaryEntity> boundary = boundaryRepository.findById(boundaryId);
+        if (boundary.isPresent()) {
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java:41]: You can replace this by just a map on the optional

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            return Optional.of(new BoundaryInfo(boundary.get().getId(), boundary.get().getFilename(), boundaryXml));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    String importBoundary(MultipartFile mpfFile) {
+        String id;
+        try {
+            Reader reader = new InputStreamReader(new ByteArrayInputStream(mpfFile.getBytes()));
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java:52]: Reader is never closed

File: src/main/java/org/gridsuite/cgmes/boundary/server/fullmodel/CgmesNames.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2017-2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.cgmes.boundary.server.fullmodel;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/cgmes/boundary/server/fullmodel/CgmesNames.java:10]: Could you mention in the javadoc of these 3 copied classes that it has been copied from powsybl core and shouls be removed in 3.6.0?

**Review by geofjamg — CHANGES_REQUESTED** [review]

**Review by geofjamg — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (FranckLecuyer) [reply pom.xml:144]: Done

File: pom.xml (hunk 2 above)
Comment (FranckLecuyer) [reply pom.xml:154]: Done

File: pom.xml (hunk 3 above)
Comment (FranckLecuyer) [reply pom.xml]: Done

File: pom.xml (hunk 4 above)
Comment (FranckLecuyer) [reply pom.xml:140]: Done

File: pom.xml (hunk 5 above)
Comment (FranckLecuyer) [reply pom.xml:190]: Done

File: pom.xml (hunk 6 above)
Comment (FranckLecuyer) [reply pom.xml]: Done

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java (hunk 7 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java]: Done

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java (hunk 8 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java]: Done

File: src/main/java/org/gridsuite/cgmes/boundary/server/fullmodel/CgmesNames.java (hunk 9 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/fullmodel/CgmesNames.java:10]: Done

Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/fullmodel/CgmesNames.java:10]: Done

**Review by geofjamg — APPROVED** [review]

**Pull request merged**